### ISSUE TITLE
TensorGraph optimizers

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#700]
I've been looking into using learning rate decay for RL, and I realized the method in #680 won't work for it.  That method relies on first creating the TensorGraph so you can get a reference to its global step variable, then using that when creating the optimizer.  But A3C requires creating the optimizer first and passing it to the constructor.  I could probably restructure it to allow that API to be changed.  But I've started wondering whether we really should just create our own set of optimizer classes for TensorGraph.  Instead of writing

```Python
optimizer = TFWrapper(tf.train.AdamOptimizer, learning_rate=0.0001)
```

you would write

```Python
optimizer = optimizers.Adam(learning_rate=0.0001)
```

Or if you wanted learning rate decay,

```Python
learning_rate = optimizers.polynomial_decay(initial=0.001, final=0.0001, steps=500000)
optimizer = optimizers.Adam(learning_rate=learning_rate)
```

It would get rid of the ugliness with `TFWrapper`, and simplify the API since users wouldn't have to deal with the `global_step` tensor.  It also would mean one less bit of explicit TensorFlow code in the API, which will be helpful if we ever decide to switch from TensorFlow to PyTorch or something else for the back end.

**Comment by rbharath** [comment]
This sounds like a great feature. I'd recommend having an abstract `Optimizer` superclass that defines the API so it will be easy for users to add on new optimizers as desired.
